utilities/scripts/bash/params.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# This script is not intended to be run by itself, and provides common functions
+# for database operations.
+
+# Check if an env var is set, if so return it,
+# if not look for it in a list of parameters and return that,
+# and if thats not found return a default
+# the default can be an individual value, or a map, in which case the default will
+# be retrieved according to the <param_name>
+# get_param <param_name> <env_var_map_name> <default_value_map> [--param1=value1 --param2=value2 ...]
```

Comment:
Mismatch name? `default_value_map`

Reply:
well not really, bash has no concept of named parameters.
So even though its assigned to a local, that's not really the name of the parameter from the callers perspective.
That said, it is better if they would align.